src/plugins/decor/index.tsx

```diff
@@ -0,0 +1,225 @@
+
+import { CDN_URL, RAW_SKU_ID, SKU_ID } from "./lib/constants";
+import { useAuthorizationStore } from "./lib/stores/AuthorizationStore";
+import { useCurrentUserDecorationsStore } from "./lib/stores/CurrentUserDecorationsStore";
+import { useUsersDecorationsStore } from "./lib/stores/UsersDecorationsStore";
+import { setOpenCreateStickerModalLazy } from "./lib/utils/requireCreateStickerModal";
+import showAuthorizationModal from "./lib/utils/showAuthorizationModal";
+import { setAvatarDecorationPreview, setDecorationGridDecoration, setDecorationGridItem } from "./ui/components";
+import { openChangeDecorationModal } from "./ui/modals/ChangeDecorationModal";
+
+let CustomizationSection;
+
+export default definePlugin({
+    name: "Decor",
+    description: "Custom avatar decorations",
+    authors: [Devs.FieryFlames],
+    patches: [
+        // Patch UserStore to include Decor avatar decorations when getting users
+        {
+            find: "getUserStoreVersion",
+            replacement: {
+                match: /(getUser=.+return )(.\[.])/,
+                replace: "$1$self.patchGetUser($2)"
+            }
+        },
+        // Patch MediaResolver to return correct URL for Decor avatar decorations
+        {
+            find: "getAvatarDecorationURL:",
+            replacement: {
+                match: /avatarDecoration,.+?;/,
+                replace: "$&const vcDecorDecoration=$self.patchGetAvatarDecorationURL(arguments[0]);if(vcDecorDecoration)return vcDecorDecoration;"
+            }
+        },
+        // Patch profile customization settings to include Decor section
+        {
+            find: "DefaultCustomizationSections",
+            replacement: {
+                match: /{user:(.)},"decoration"\),/,
+                replace: "$&$self.DecorSection(),"
+            }
+        },
+        // Obtain CustomizationSection component
+        {
+            find: "e.titleIcon",
+            replacement: {
+                match: /function (\i)\(\i\){var \i,\i=\i\.title/,
+                replace: "$self.CustomizationSection=$1;$&"
+            }
+        },
+        // Decoration modal module
+        {
+            find: ".decorationGridItem",
+            replacement: [{
+                match: /(?:,)((\i)=function\(.\){var \i=\i\.children)/,
+                replace: ";var $2;$self.DecorationGridItem=$1"
+            },
+            {
+                match: /const (\i)=(function\(\i\){var \i=\i\.user,\i=\i\.avatarDecorationOverride,\i=\i\.className)/,
+                replace: "let $1;$1=$self.AvatarDecorationPreview=$2"
+            },
+            {
+                match: /,(\i)=(function\(\i\){var \i=\i\.user,\i=\i\.avatarDecoration,)/,
+                replace: ";var $1;$self.DecorationGridDecoration=$1=$2"
+            },
+            {
+                match: /\i\.\i\.isItemViewed\((\i)\)/,
+                replace: "$1.skuId !== $self.SKU_ID ? $& : true"
```

Comment:
i'd do this just in case
```suggestion
                replace: "($1.skuId !== $self.SKU_ID ? $& : true)"
```